conda_build/source.py

```diff
@@ -612,7 +612,7 @@ def patch_or_reverse(patch, patch_args, cwd, stdout, stderr):
 
     exception = None
     if not isfile(path):
-        sys.exit('Error: no such patch: %s' % path)
+        raise RuntimeError('Error: no such patch: %s' % path)
```

Comment:
You can raise `SystemExit` and keep "similar" to what happened before with a nicer way to catch. However, I agree that this is more like a run time error instead.

Reply:
Right and agreed a `RuntimeError` is better. The issue is if a user is catching errors, they cannot use `Exception` since `SystemExit` doesn't inherit from that. Asking users to catch `SystemExit` is dangerous and definitely a python anti-pattern.